include/revng/Model/schema.yml

```diff
@@ -0,0 +1,701 @@
+
+  - name: RegisterState
+    type: enum
+    members:
+      - name: "No"
+      - name: NoOrDead
+      - name: Dead
+      - name: "Yes"
+      - name: YesOrDead
+      - name: Maybe
+      - name: Contradiction
+
+  - name: TypeKind
+    doc: Enum for identifying different kind of model types
+    type: enum
+    members:
+      - name: Primitive
+      - name: Enum
+      - name: Typedef
+      - name: Struct
+      - name: Union
+      - name: CABIFunctionType
+      - name: RawFunctionType
+
+  - name: PrimitiveTypeKind
+    type: enum
+    members:
+      - name: Void
+      - name: Generic
+      - name: PointerOrNumber
+      - name: Number
+      - name: Unsigned
+      - name: Signed
+      - name: Float
+
+  - name: QualifierKind
+    doc: |
+      Enum for identifying different kinds of qualifiers
+
+      Notice that we are choosing to represent pointers and arrays as qualifiers.
+      The idea is that a qualifier is something that you can add to a type T to
+      obtain another type R, in such a way that if T is fully known also R is fully
+      known. In this sense Pointer and Array types are qualified types.
+    type: enum
+    members:
+      - name: Pointer
+      - name: Array
+      - name: Const
+
+  - name: FunctionEdgeType
+    doc: Type of edge on the CFG
+    type: enum
+    members:
+      - doc: Branch due to function-local CFG (a regular branch)
+        name: DirectBranch
+      - doc: A call to a fake function
+        name: FakeFunctionCall
+      - doc: A return from a fake function
+        name: FakeFunctionReturn
+      - doc: A function call for which the cache was able to produce a summary
+        name: FunctionCall
+      - doc: A function call for which the target is unknown
+        name: IndirectCall
+      - doc: A proper function return
+        name: Return
+      - doc: A branch returning to the return address, but leaving the stack in an unexpected situation
+        name: BrokenReturn
+      - doc: A branch representing an indirect tail call
+        name: IndirectTailCall
+      - doc: A branch representing a longjmp or similar constructs
+        name: LongJmp
+      - doc: A killer basic block (killer syscall or endless loop)
+        name: Killer
+      - doc: The basic block ends with an unreachable instruction
+        name: Unreachable
+
+  - name: FunctionType
+    type: enum
+    members:
+      - doc: A function with at least one return instruction
+        name: Regular
+      - doc: A function that never returns
+        name: NoReturn
+      - doc: A function with at least one non-proper return instruction. This typically represents outlined function prologues.
+        name: Fake
+
+  # Structs
+  - name: Type
+    doc: Base class of model types used for LLVM-style RTTI
+    type: struct
+    fields:
+      - name: Kind
+        type: model::TypeKind::Values
+        default: TypeKind::Invalid
+      - name: ID
+        type: uint64_t
+        default: "0"
```

Comment:
Is there a single situation where `default` is not `0`/`Invalid`?
I'd like to drop `default` or at least make it optional as much as possible.
Initializing fields with `{}` should do the trick.

Reply:
I managed to remove all defaults from the yaml!
Turns out it possible to initialize all fields like `FieldType Name = FieldType{}`. Unfortunately, some classes (e.g. `MetaAddress`) have an explicit default constructor, so it's not possible to just write `FieldType Name = {}`, hence the requirement to write the field name twice. I could use `auto`, I don't know if you like it or not.
Since it is autogenerated code it does not matter that much one way or the other IMO, and I find the repetition better than using auto, at least all type names can be found in the code at the same column, I find it easier to read.

Reply:
The following should work:
```c++
MetaAddress Asd = MetaAddress {};
```

Reply:
That's what I did :)